README.md

```diff
@@ -20,22 +19,66 @@ Searvey aims to provide the following functionality:
 
 ## Installation
 
-The package can be installed with `conda`:
+The package can be installed with `pip`:
+
+```
+pip install searvey
+```
+
+and conda`:
+
+```
+conda install -c conda-forge searvey
+```
 
-`conda install -c conda-forge searvey`
 
 ## Development
 
+In order to develop `searvey` you will need:
+
+- Python 3.8+
+- GNU Make
+- [poetry](https://python-poetry.org/) >= 1.2 (you can install it with [pipx](https://github.com/pypa/pipx): `pipx install poetry`).
+- [poetry-dynamic-versioning](https://github.com/mtkennerly/poetry-dynamic-versioning) which is a poetry plugin.
+  Take note that this needs to be installed in the same (virtual) environment as poetry, not in the `searvey` one!
+  If you used `pipx` for installing `poetry`, then you can inject it in the proper env with `pipx inject poetry poetry-dynamic-versioning`.
+- [pre-commit](https://pre-commit.com/). You can also install this one with `pipx`: `pipx install pre-commit`
+
+In order to setup the dev environment you can use:
+
 ```
 python3 -mvenv .venv
 source .venv/bin/activate
-poetry install
-pre-commit install
+make init
+```
+
+which will:
+
+1. create and activate a virtual environment,
+2. install the full set of dependencies
+3. Setup the pre-commit hooks
+
+After that you should run the tests with:
+
+```
+make test
 ```
 
+If you execute `make` without arguments, you should see more subcommands. E.g.
+
+```
+make mypy
+make lint
+make docs
+make deps
+```
```

Comment:
All of this is done as a part of precommit (when we execute `git commit ...`) when hooks are installed, right?

Reply:
If you do `pre-commit install`, which gets executed when you do `make init`, then indeed these checks will run. Nevertheless when you are developing, you usually want to have a way of running these tests individually, too. Running some of the checks manually involves some boiler-plate code. E.g. to run `ruff` you need `pre-commit run ruff -a`. That's where the Makefile comes into place. It defines easy to use/discover aliases

Reply:
I see, I just wanted to make sure I didn't miss anything. From the readme I thought it meant that the precommits don't execute automatically.